src/pkg/artifact/dao/dao.go

```diff
@@ -79,29 +59,39 @@ func New() DAO {
 type dao struct{}
 
 func (d *dao) Count(ctx context.Context, query *q.Query) (int64, error) {
-	if query != nil {
-		// ignore the page number and size
-		query = &q.Query{
-			Keywords: query.Keywords,
-		}
-	}
-	qs, err := querySetter(ctx, query)
-	if err != nil {
-		return 0, err
+	model := &Artifact{}
+
+	options := []selector.Option{
+		selector.From(model),
+		selector.Filter(model, query.GetKeyWords()),
+		filterBase(query),
+		filterTag(query),
+		filterLabel(query),
 	}
-	return qs.Count()
+
+	return selector.New(ctx, options...).Count(ctx)
 }
+
 func (d *dao) List(ctx context.Context, query *q.Query) ([]*Artifact, error) {
-	artifacts := []*Artifact{}
-	qs, err := querySetter(ctx, query)
-	if err != nil {
-		return nil, err
-	}
-	if _, err = qs.All(&artifacts); err != nil {
-		return nil, err
+	model := &Artifact{}
+
+	options := []selector.Option{
+		selector.From(model),
```

Comment:
Seems every function call for `selector` we need to pass the `model` as a parameter, is it possible to avoid this?

Reply:
Although we only use `selector` in the dao of artifacts that using the `selector.From`, but it also can work with raw SQL. For example,

```go
func undeleted (b, c) nil {
     b.Where( c.IsNull("deleted_at",))
}

selector.New(ctx, func(b, c) nil {
   b.From("tablea").Where("col1", c.E("c1"))
},
undeleted,
)
```

selector.From is just a helper to apply `FROM tablename` for the model.